test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -165,6 +165,37 @@ public void ParseNotControlNorWhitespace ()
             Assert.Throws<ParseException>(() => Parsers.NotControlNorWhitespace(EXCEPT_CHARS).End().Parse("$"));
         }
 
+        [Fact]
```

Comment:
This helped me experiment my way into the solution, and seemed helpful to keep around, but I'm not attached to it, as it is perhaps a bit confusing to leave around. Thoughts on keeping vs dropping?

Reply:
I guess if we go in some detail about the checks there's no reason to drop them generally.

I will have a closer look at them later, just reading through I don't get the intention of each check.